### DISCUSSION TITLE
Enable/Disable ssh keys

### DISCUSSION THREAD

**Discussion by jikkujose** [community/community#82748] [Code Security]
Implement enabling/disabling of ssh keys so that, locking out a key in case of device theft is trivial.

**Comment by rishiskhare** [comment]
## 1. Create a script to manage SSH keys:

Create a shell script, let's call it manage_ssh_keys.sh. This script will handle enabling and disabling of SSH keys.

KEY_FILE="$HOME/.ssh/authorized_keys"

function enable_key() {
    local key="$1"
    grep -q "$key" "$KEY_FILE" || echo "$key" >> "$KEY_FILE"
    echo "Key enabled: $key"
}

function disable_key() {
    local key="$1"
    sed -i "/$key/d" "$KEY_FILE"
    echo "Key disabled: $key"
}

## 2. Make the script executable:
chmod +x manage_ssh_keys.sh

## 3. Enable or Disable SSH Keys:
Enable a key:
./manage_ssh_keys.sh enable "ssh-rsa AAAA... user@example.com"

Disable a key:
./manage_ssh_keys.sh disable "ssh-rsa AAAA... user@example.com"

**Comment by krabelize** [comment]
You can also setup a centralized SSH key management system that allows you to easily manage and control SSH keys across your organization. This system should provide features for enabling, disabling, and revoking SSH keys as needed. Please mark this as a valid answer if this helped you.
